src/librustc_ast_passes/ast_validation.rs

```diff
@@ -931,14 +940,14 @@ impl<'a> Visitor<'a> for AstValidator<'a> {
             ItemKind::Union(ref vdata, _) => {
                 if let VariantData::Tuple(..) | VariantData::Unit(..) = vdata {
                     self.err_handler()
-                        .span_err(item.span, "tuple and unit unions are not permitted");
+                        .span_err(item.ident.span, "tuple and unit unions are not permitted");
```

Comment:
These `.span` -> `.ident.span` changes makes it feel too "naked"; I would like to avoid changing them in this PR at least.